### DISCUSSION TITLE
Exended MIDI Objects dialog

### DISCUSSION THREAD

**Discussion by willeke1234** [GrandOrgue/grandorgue#2020] [General]
I'm playing around with the MIDI Objects dialog to see what wxWidgets can do. One thing leads to another and now I have this:

![Scherm­afbeelding 2024-10-09 om 13 16 28](https://github.com/user-attachments/assets/6b95bc11-7273-4fd0-a023-69f018534e1d)

and

![Scherm­afbeelding 2024-10-09 om 13 51 15](https://github.com/user-attachments/assets/f1757cfd-6c2f-43dc-8477-0b1b16326965)

It’s a work in progress. Would this be a useful addition? Is it clear? Too much? What do you think so far?

**Comment by larspalo** [comment]
I personally think that improving the Initial MIDI tab (for instance adding all available built-in elements) is a much more important task.

Also, in the MIDI-Object dialog, there is a slight problem with some objects shown (for instance Stops/Ranks) that are not of an Input function - which means that they are actually controlled by a switch (Drawstop) instead. However, with such changes that you've tested, one can indeed have a clearer overview of what MIDI configurations are actually set for a certain sample set and I can see that it perhaps could have some merits in that way.

**Comment by willeke1234** [comment]
>I personally think that improving the Initial MIDI tab (for instance adding all available built-in elements) is a much more important task.

May I ask why? And what else should be changed?

**Reply by kerkovits** [reply]
For example, I use MIDI events to trigger the built-in couplers accessible in the Panels menu. Now I have to repeat the same configuration for each organ, which is quite tedious.

**Reply by kerkovits** [reply]
See also https://github.com/GrandOrgue/grandorgue/issues/1974

**Reply by larspalo** [reply]
@willeke1234 Exactly as @kerkovits say. The initial MIDI will apply to any new organ and immediately set it up according to the user choices. If all the built-in elements were available there a lot of repetition could be completely avoided when configuring a sample set for a certain console.

**Reply by willeke1234** [reply]
It sounds that the MIDI settings of the built-in elements are stored in the wrong place. I don't have the knowledge to improve the Initial MIDI settings or to fix this design flaw.
